theano/gpuarray/opt.py

```diff
@@ -751,38 +751,27 @@ def local_gpua_elemwise(op, context_name, inputs, outputs):
         gpu_output = res(*new_inputs)
         return [gpu_output]
     elif op.scalar_op in (scalar.add, scalar.mul):
-        return split_huge_add_or_mul(outputs[0].owner, res).outputs
+        try:
+            return [split_inputs(inputs, max_inputs_to_GpuElemwise(outputs), res)]
+        except ValueError:
+            return False
     else:
         return res
 
 
-def split_huge_add_or_mul(node, op=None):
-    """
-    For add and mul, it can happen that we have too much input
-    That will make nvcc fail compilation of our current code.
-    We don't want node in the graph that can't execute
-    as this break DebugMode.
+def split_inputs(inputs, max_nb_inputs, op):
```

Comment:
Do not drop the docstring, adapt it and add definitions of parameters, namely for ```op```
```
Parameters
----------
op : Theano operator instance
     Operator that should be used to rebuild the computation graph with smaller number of inputs per node.
```